### PR TITLE
Fix build script (K-1811)

### PR DESCRIPTION
**Pull request by plutonik-a**
This PR incorporates commits from PR https://github.com/HistoryAtState/hsg-project/pull/56:  
* Add facility for working with Forks
    ```
       # If the `upstream` and `origin` are different then both will
       # be configured, otherwise only the origin will be configured.
       # If you work with a fork, you can set that as the `origin`.
       git.public.remotes.upstream.org=https://github.com/HistoryAtState
       git.public.remotes.origin.org=https://github.com/HistoryAtState
* Replace `foreach-js` target with `ant-contrib` `foreach` function.

Note: In a first attempt to support the required JavaScript engine for Java 15+, I added 21a8f75ddd042d07913951374c5b9c280283d5b6, but reverted it later as gives no permanent fix when re-installing the project again.

### CODE REVIEW

**Review by yamahito — APPROVED** [review]
Looks good - building really quickly.

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ To automate the following steps:
 ## Prerequisites
 
 - eXist 5.x
-- Current versions of ant, git, and bower executables
+- Current versions of ant, ant-contrib, git, and bower executables
```
Comment (joewiz) [inline README.md]: The ant-contrib dependency is already handled by the pom.xml. It shouldn't be listed here as a prerequisite.

File: build.xml [hunk 2]
```diff
@@ -459,34 +513,36 @@
...
-              antc.perform();
-            }
-        ]]>
-    </scriptdef>
-
     <target name="git.clone">
-        <echo message="Cloning ${item}..."/>
-        <exec executable="${git}" outputproperty="git.output" dir="${repos}">
+        <echo message="Cloning ${origin}..."/>
+        <exec executable="${git}" outputproperty="git.output" dir="${repos}" failonerror="true">
```
Comment (joewiz) [inline build.xml]: Is this newly added attribute required? In my testing, it prevents us from re-running "1. Clone all repositories & resources." Our maintenance steps depend on the ability to re-run this, to fetch new maven dependencies and even possibly clone newly added repositories. The previous error condition was totally harmless. It returns the following for each repository that already exists:

```
git.clone:
     [echo] Cloning https://github.com/HistoryAtState/hsg-shell.git...
     [exec] Result: 128
     [echo] fatal: destination path 'hsg-shell' already exists and is not an empty directory.
```

I'm open to other solutions, but this change - as is - breaks our system.

File: build.xml [hunk 3]
```diff
@@ -798,28 +854,28 @@
 
     <target name="ping-localhost">
         <condition property="exist-running">
-            <http requestmethod="HEAD" url="http://localhost:8080/exist/status"/>
+            <http requestmethod="HEAD" url="http://localhost:8080/exist"/>
```
Comment (joewiz) [inline build.xml]: Could you explain the need for this change? This change forces Ant to follow a 302 redirect. Previously, the request would return a 200, with no need for following a redirect.

**Comment by joewiz** [comment]
I've finished my review of this PR in its current state. The only questions I have are as listed above.

File: README.md (hunk 1 above)
Comment (plutonik-a) [reply README.md]: Fixed with https://github.com/evolvedbinary/hsg-project/pull/3/commits/6037e2b3dc231229baf50d6e691128e1fa6d1a60

File: build.xml (hunk 3 above)
Comment (plutonik-a) [reply build.xml]: Fixed with https://github.com/evolvedbinary/hsg-project/pull/3/commits/1a66bca72b4bfd5acd9d01a8d022a645cfddafd1

File: build.xml (hunk 2 above)
Comment (plutonik-a) [reply build.xml]: Fixed with https://github.com/evolvedbinary/hsg-project/pull/3/commits/d10ec0134b3b7247f9fdd9ce9a58518bc390bd50:
I have set a dynamic `failonerror` attribute value, achieved by pre-checking if the repo is existing or not. 
The build will now only fail on error, if the respective repo is not yet existing. If it is existing and an error occurs, the build will finish and display useful error messages.

Comment (joewiz) [reply build.xml]: @plutonik-a Thank you! I see the updated steps:

```text
check-if-repo-exists:
     [echo] The repo 'hsg-shell' already exists and cannot be overwritten. Please delete this repo and try again.
```

I think the 2nd sentence could be softened to something like:

> If you need to re-clone this repo, please delete 'hsg-shell' and try again.

**Review by joewiz — APPROVED** [review]
Looks and works great, thank you!

**Pull request merged**